src/modm/platform/can/sam-mcan/can.hpp

```diff
@@ -0,0 +1,434 @@
+
+public:
+	// Expose template parameters to be checked by e.g. drivers or application
+	static constexpr size_t RxBufferSize = std::min(MessageRam::RxFifo0Elements, MessageRam::RxFifo1Elements);
+	static constexpr size_t TxBufferSize = MessageRam::TxFifoElements;
+
+	template<class... Pins>
+	static void
+	connect(modm::platform::InputType inputType = modm::platform::InputType::Floating)
+	{
+		using RxPin = GetPin_t<PeripheralPin::Rx, Pins...>;
+		using TxPin = GetPin_t<PeripheralPin::Tx, Pins...>;
+
+		using Peripheral = typename Peripherals::Mcan<id>;
+
+		if constexpr (!std::is_void_v<RxPin>) {
+			RxPin::configure(inputType);
+			using RxConnector = typename RxPin::template Connector<Peripheral, typename Peripheral::template Canrx<id>>;
+			RxConnector::connect();
+		}
+
+		if constexpr (!std::is_void_v<TxPin>) {
+			using TxConnector = typename TxPin::template Connector<Peripheral, typename Peripheral::template Cantx<id>>;
+			TxConnector::connect();
+		}
+	}
+
+	/**
+	 * Enables the clock for the CAN controller and resets all settings
+	 *
+	 * \tparam SystemClock
+	 * 			System clock struct with an MCAN member containing
+	 * 			the clock speed supplied to the peripheral.
+	 * 			\warning	The CAN subsystem prescaler can be configured using
+	 * 						the RCC module and must be taken into account in
+	 * 						the provided clock speed value.
+	 * \tparam bitrate
+	 * 			Nominal CAN bitrate
+	 * \tparam tolerance
+	 * 			Maximum relative deviation between requested and resulting
+	 * 			CAN bitrates. If the tolerance is exceeded a compile-time
+	 * 			assertion will be triggered.
+	 * \tparam fastDataBitrate
+	 * 			CAN bitrate for data in FD frames with bit rate switching
+	 * 			Set to 0 to disable CANFD support.
+	 *
+	 * \param interruptPriority
+	 * 			Not used in this driver: Interrupt vector priority (0=highest to 15=lowest)
+	 * \param startupMode
+	 * 			Mode of operation set after initialization
+	 * 			\see Mcan<id>::Mode
+	 * \param overwriteOnOverrun
+	 * 			Once a receive FIFO is full the next incoming message
+	 * 			will overwrite the previous one if \c true otherwise
+	 * 			the incoming message will be discarded
+	 *
+	 * \warning	Has to called after connect(), but before any
+	 * 			other function from this class!
+	 */
+	template<
+		class SystemClock,
+		bitrate_t bitrate=kbps(125),
+		percent_t tolerance=pct(1),
+		bitrate_t fastDataBitrate=0 // 0: MCAN mode disabled
+	>
+	static inline void
+	initialize(uint32_t interruptPriority, Mode startupMode = Mode::Normal,
+				bool overwriteOnOverrun = true)
+	{
+		using Timings = CanBitTiming<
+			SystemClock::Mcan,
+			bitrate,
+			9, 8, 7, 7
+		>;
+		Timings::template assertBitrateInTolerance<tolerance>();
+
+		std::optional<CanBitTimingConfiguration> fastDataTimings = std::nullopt;
+
+		if constexpr (fastDataBitrate != 0) {
+			using DataTimings = CanBitTiming<
+				SystemClock::Mcan,
+				fastDataBitrate,
+				5, 5, 4, 4
```

Comment:
```suggestion
				5, 5, 4, 3
```
`SJW` is only 3 bits but the value is hard coded to 1 in the calculator anyway.